limitador/src/storage/atomic_expiring_value.rs

```diff
@@ -0,0 +1,135 @@
+use std::sync::atomic::Ordering;
+use std::sync::atomic::{AtomicI64, AtomicU64};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+#[derive(Debug)]
+pub(crate) struct AtomicExpiringValue {
+    value: AtomicI64,
+    expiry: AtomicU64, // in microseconds
+}
+
+impl AtomicExpiringValue {
+    pub fn new(value: i64, expiry: SystemTime) -> Self {
+        let expiry = expiry
+            .duration_since(UNIX_EPOCH)
+            .expect("SystemTime before UNIX EPOCH!")
+            .as_micros() as u64;
+        Self {
+            value: AtomicI64::new(value),
+            expiry: AtomicU64::new(expiry),
+        }
+    }
+
+    pub fn value_at(&self, when: SystemTime) -> i64 {
+        let when = when
+            .duration_since(UNIX_EPOCH)
+            .expect("SystemTime before UNIX EPOCH!")
+            .as_micros() as u64;
+        let expiry = self.expiry.load(Ordering::SeqCst);
+        if expiry <= when {
+            return 0;
+        }
+        self.value.load(Ordering::SeqCst)
+    }
+
+    pub fn value(&self) -> i64 {
+        self.value_at(SystemTime::now())
+    }
+
+    pub fn update(&self, delta: i64, ttl: u64, when: SystemTime) {
```

Comment:
With these changes, when we get rid of the RwLock guarding concurrent updates, we will want to return the counter's `value` here and probably recheck it against the Limit's ... We probably might want to discuss this at some point.